lib/tpm2_alg_util.c

```diff
@@ -330,3 +330,57 @@ bool pcr_parse_digest_list(char **argv, int len,
 
     return true;
 }
+
+UINT8* tpm2_extract_plain_signature(UINT16 *size, TPMT_SIGNATURE *signature) {
+
+    UINT8 *buffer = NULL;
+    *size = 0;
+
+    switch (signature->sigAlg) {
+    case TPM_ALG_RSASSA:
+        *size = sizeof((*signature).signature.rsassa.sig.t.buffer);
```

Comment:
I would argue that using  the arrow operator is a much common idiom than doing a pointer dereference here, i.e:

```
*size = sizeof(signature->signature.rsassa.sig.t.buffer);
```
This comment is also for all the places in the code that do the same.